src/linear_algebra/finite_dimensional.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2019 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+
+Definition and basic properties of finite dimensional vector spaces.
+
+The class `finite_dimensional` is defined to be `is_noetherian`, for ease of transfer of proofs.
+However an additional constructor `finite_dimensional.of_span_finite_eq_top` is provided to prove
+finite dimensionality in a conventional manner.
+
+Also defined is `findim`, the dimension of a finite dimensional space, returning a `nat`,
+as opposed to `dim`, which returns a `cardinal`,
+-/
+
+import ring_theory.noetherian linear_algebra.dimension
+import ring_theory.principal_ideal_domain
+
+universes u v w
+
+open vector_space cardinal submodule module function
+
+variables {K : Type u} {V : Type v} [discrete_field K] [add_comm_group V] [vector_space K V]
+
+/-- `finite_dimensional` vector spaces are defined to be noetherian modules.
+  Use `finite_dimensional.of_span_finite_eq_top` to prove finite dimensional from a conventional
+  definition. -/
+@[reducible] def finite_dimensional (K V : Type*) [discrete_field K]
+  [add_comm_group V] [vector_space K V] := is_noetherian K V
+
+namespace finite_dimensional
+
+lemma finite_of_linear_independent {ι : Type w} [decidable_eq V] [decidable_eq ι]
```

Comment:
I think something like this is true for arbitrary noetherian modules, right. How much of this file can be done in the general setting?

Reply:
I don't think it is true, though I can't think of a counter example. Just because a submodule has a finite basis doesn't necessarily mean that every basis for that submodule is finite? Otherwise why is the definition of noetherian not just that the module is finitely generated, rather than every submodule? I'd be interested to know the counterexample.

Reply:
If you have an infinite basis you can create an infinite increasing chain of submodules:
span {v_1} \subset span {v_1, v_2} \subset ...
this contradicts being noetherian.

Reply:
Of course. This is quite a hard proof.

Reply:
Generalized to noetherian modules